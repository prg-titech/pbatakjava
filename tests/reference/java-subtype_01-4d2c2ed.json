{
    "basename": "java-subtype_01-4d2c2ed",
    "cmd": "{infile}",
    "infile": "tests/subtype_01/build.sh",
    "infile_hash": "75687d00d82803ba735d04c9e5dd0aa532fa06b68b45ea6aec4b3598",
    "stdout": "java-subtype_01-4d2c2ed.stdout",
    "stdout_hash": "1a40102bd6632b4d01bdf53054254f4a216a3d73f03ccbd6b9975fba",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}