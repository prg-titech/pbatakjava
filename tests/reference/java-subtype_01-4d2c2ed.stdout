
public interface Point {
}

public interface Point_Factory {
  public Point make(int x, int y);
}

public class Point_v1 implements Point {
  public int x;

  public int y;

  public Point_v1(int x, int y) {
    this.x = x;
    this.y = y;
  }
}

public class Point_v1_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v1(x, y);
  }
}

public class Point_v2 implements Point {
  public int x;

  public int y;

  public Point_v2(int x, int y) {
    this.x = x;
    this.y = y;
  }
}

public class Point_v2_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v2(x, y);
  }
}

public interface ColoredPoint {
}

public interface ColoredPoint_Factory {
  public ColoredPoint make(int a, int b, int red, int green, int blue);
}

public class ColoredPoint_v1 extends Point_v1 implements ColoredPoint {
  public int red;

  public int green;

  public int blue;

  public ColoredPoint_v1(int a, int b, int red, int green, int blue) {
    super(a, b);
    this.red = red;
    this.green = green;
    this.blue = blue;
  }
}

public class ColoredPoint_v1_Factory implements ColoredPoint_Factory {
  public ColoredPoint make(int a, int b, int red, int green, int blue) {
    return new ColoredPoint_v1(a, b, red, green, blue);
  }
}





public class Test {
  public static void main(String[] args) {
    Point_v1 p1 = new Point_v1(1, 2);
    Point_v1 p2 = new ColoredPoint_v1(1, 2, 3, 4, 5);
  }
}
